### PR TITLE
Updating docs-server docs and making it more reliable.

### PR DESCRIPTION
**Pull request by robotrapta**
Updating `start_docs_server.sh` with README instructions and making it work more broadly.

### CODE REVIEW

File: docs/start_docs_server.sh [hunk 1]
```diff
@@ -11,13 +36,8 @@ while true; do
...
   sleep 15
   
-  if [[ "$OSTYPE" == "darwin"* ]]; then
-    CHANGED_FILE=$(fswatch -1 --exclude 'docs/static/api-reference-docs' --exclude 'build/' --exclude '/docs/.docusaurus' --exclude 'changes.log' --exclude 'docs/node_modules/.cache/webpack' --exclude '.git/'  $WATCH_PATH)
-    echo "Detected changes in: $CHANGED_FILE"
-  else
-    echo "OS not supported"
-    exit 1
-  fi
+  CHANGED_FILE=$(fswatch -1 --exclude 'docs/static/api-reference-docs' --exclude 'build/' --exclude '/docs/.docusaurus' --exclude 'changes.log' --exclude 'docs/node_modules/.cache/webpack' --exclude '.git/'  $WATCH_PATH)
```
Comment (sunildkumar) [inline docs/start_docs_server.sh:39]:
```suggestion
  CHANGED_FILE=$(fswatch -1 --exclude 'docs/static/api-reference-docs' --exclude 'build/' --exclude '/docs/.docusaurus' --exclude 'docs/node_modules/.cache/webpack' --exclude '.git/'  $WATCH_PATH)
```

Comment (sunildkumar) [inline docs/start_docs_server.sh:39]:
```suggestion
  CHANGED_FILE=$(fswatch -1 --exclude 'docs/static/api-reference-docs' --exclude 'build/' --exclude '/docs/.docusaurus' --exclude 'docs/node_modules/.cache/webpack' --exclude '.git/'  $WATCH_PATH)
```

Comment (sunildkumar) [reply docs/start_docs_server.sh:39]: the diff is weird but trying to not exclude changes.log as that file no longer is created by the script.

**Comment by sunildkumar** [comment]
@robotrapta This was ready to go but I think we forgot to merge it!

**Comment by robotrapta** [comment]
I fixed all the merge conflicts by manually re-basing.  So I think this is ready again @sunildkumar

**Review by blaise-muhirwa — APPROVED** [review]
Looks great!

**Pull request merged**